jme3-examples/src/main/java/jme3test/light/TestLightControlSpot.java

```diff
@@ -0,0 +1,134 @@
+        al.setColor(ColorRGBA.White.mult(2f));
+        rootNode.addLight(al);
+
+        spot = new SpotLight();
+
+        spot.setSpotRange(1000);
+        spot.setSpotInnerAngle(5 * FastMath.DEG_TO_RAD);
+        spot.setSpotOuterAngle(10 * FastMath.DEG_TO_RAD);
+        spot.setColor(ColorRGBA.White.mult(10));
+        rootNode.addLight(spot);
+
+
+//        PointLight pl=new PointLight();
+//      pl.setPosition(new Vector3f(77.70334f, 34.013165f, 27.1017f));
+//      pl.setRadius(1000);
+//      pl.setColor(ColorRGBA.White.mult(2));
+//      rootNode.addLight(pl);
+        lightMdl = new Geometry("Light", new Dome(Vector3f.ZERO, 2, 32, 5, false));
+        lightMdl.setMaterial(assetManager.loadMaterial("Common/Materials/RedColor.j3m"));
+        lightMdl.setLocalTranslation(new Vector3f(0, 0, 0));
+        lightMdl.setLocalRotation(new Quaternion().fromAngles(FastMath.PI / 2F, 0, 0));
+        rootNode.attachChild(lightMdl);
+
+        /*
+         * We need this Dome doesn't have a "floor."
+         */
+        Geometry lightFloor = new Geometry("LightFloor", new Cylinder(2, 32, 5, .1F, true));
+        lightFloor.setMaterial(assetManager.loadMaterial("Common/Materials/RedColor.j3m"));
+        lightFloor.getMaterial().setColor("Color", ColorRGBA.White);
+//        lightFloor.setLocalRotation(new Quaternion().fromAngles(FastMath.PI / 2F, 0, 0));
+
+        lightNode = new Node();
+        lightNode.addControl(new LightControl(spot));
+//        lightNode.setLocalTranslation(-5, 0, 0);
+        lightNode.attachChild(lightMdl);
+        lightNode.attachChild(lightFloor);
+        rootNode.attachChild(lightNode);
+    }
+
+    public void setupDome() {
+        Geometry dome = new Geometry("Dome", new Sphere(16, 32, 30, false, true));
+        dome.setMaterial(new Material(this.assetManager, "Common/MatDefs/Light/PBRLighting.j3md"));
+        dome.setLocalTranslation(new Vector3f(0, 0, 0));
+        rootNode.attachChild(dome);
+    }
+
+    @Override
+    public void simpleInitApp() {
+        flyCam.setMoveSpeed(30);
+
+        setupLighting();
+        setupDome();
+    }
+
+    @Override
+    public void simpleUpdate(float tpf) {
+        /*
+         * In Radians per second
+         */
+        final float ROT_SPEED = FastMath.PI * 2;
+        /*
+         * 360 degree rotation
+         */
+        final float FULL_ROT = FastMath.PI * 2;
+        angles[0] += rotAxis.x * ROT_SPEED * tpf;
+        angles[1] += rotAxis.y * ROT_SPEED * tpf;
+        angles[2] += rotAxis.z * ROT_SPEED * tpf;
+        lightNode.setLocalRotation(new Quaternion().fromAngles(angles));
+        super.simpleUpdate(tpf);
+        /*
+         * Make sure they are equal.
+         */
+        if (spot.getPosition().subtract(lightNode.getWorldTranslation()).lengthSquared() > 0.1F) {
+            System.err.printf("Translation not equal: is %s (%s), needs to be %s\n", lightNode.getWorldTranslation(), lightNode.getLocalTranslation(), spot.getPosition());
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.